app/controllers/GreeterServiceImpl.scala

```diff
@@ -1,16 +1,20 @@
 package controllers
 
+import java.lang.annotation.Annotation
+
 import akka.NotUsed
 import akka.stream.Materializer
 import akka.stream.scaladsl.Source
 import io.grpc.examples.helloworld.{GreeterService, HelloReply, HelloRequest}
-import javax.inject.{Inject, Singleton}
+import javax.inject.{Inject, Named, Singleton}
 
 import scala.concurrent.Future
 
 /** Would be written by the user, with support for dependency injection etc */
 @Singleton
-class GreeterServiceImpl @Inject()(implicit mat: Materializer) extends GreeterService {
+@Named("grpc-impl")
```

Comment:
`@Named` needs to go where it is injected, not here.

Reply:
To better explain:

```scala
// bind it using named
bind[GreeterService].qualifiedWith("grpc-impl").to[GreeterServiceImpl]

// and when injecting
class Foo @Inject()(@Named("grpc-impl") greeterService: GreeterService)
```

Reply:
Yeah I guess so - though https://dzone.com/articles/java-ee6-cdi-named-components also mentions it on implementation classes